packages/template-retail-react-app/app/hooks/use-einstein.js

```diff
@@ -440,51 +429,82 @@ const useEinstein = () => {
         return reco
     }
 
+    const getEventUserParameters = async () => {
+        const usid = await getUsidWhenReady()
+        const userParameters = {}
+
+        // If the shopper is authenticated, send the encrypted user ID so recommendations
+        // persist across logins
+        if (isRegistered) {
+            userParameters.userId = await getEncUserIdWhenReady()
+        }
+
+        userParameters.cookieId = usid
+
+        return userParameters
+    }
```

Comment:
```suggestion
    const getEventUserParameters = async () => {
        return {
            cookieId: await getUsidWhenReady(),
            userId: isRegistered ? await getEncUserIdWhenReady() : undefined
        }
    }
```

Reply:
(not tested)

Reply:
I just tested this. It works!

I was wondering if `userId: undefined` would get set in the object (this would be a problem if it did) but it doesn't seem to be the case